src/ripple/crypto/impl/ECIES.cpp

```diff
@@ -97,6 +99,11 @@ static void getECIESSecret (const openssl::ec_key& secretKey, const openssl::ec_
     memset (hbuf, 0, ECIES_KEY_LENGTH);
 }
 
+static void getECIESSecret (uint256 const& secretKey, Blob const& publicKey, ECIES_ENC_KEY_TYPE& enc_key, ECIES_HMAC_KEY_TYPE& hmac_key)
+{
```

Comment:
Well neither of them work very well because we have a 'unity' build. multiple .cpp files are `#include`ed together to form a big translation unit. This can break the isolation of static functions and anonymous namespaces. So code with heavy dependence on said constructs is discouraged.
See:
http://buffered.io/posts/the-magic-of-unity-builds/
https://cheind.wordpress.com/2009/12/10/reducing-compilation-time-unity-builds/